designs/exportable-subflow/README.md

```diff
@@ -24,28 +24,52 @@ are in the roadmap. (*TODO: add a link to the design doc once added*).
 
 ### Details
 
-#### Exporting Subflow as Node
-
-Node can be exported as JSON code or converted to a Node.
+Node can be exported as JSON code (*SUBFLOW Module*) or converted to a Node (*SUBFLOW Node*). In this document, design for *SUBFLOW Module* is described.
 
+#### Exporting SUBFLOW Module
 
 ![Subflow-export-UI](Subflow-export-UI.png)
 
-By clicking **Export** button on Subflow Setting Panel of Subflow template, Subflow Export Panel is shown.
-
-In this panel, new node information such as node name, version, etc. can be specified. There is **As JSON** check box on this panel.  If this is checked, JSON export panel is shown.  Otherwise, new node is installed on  editor palette.  
+By clicking **Export SUBFLOW** button on SUBFLOW template setting panel, SUBFLOW export panel is shown.
 
-New node created from SUBFLOW is sealed.  Thus, its internal flow or UI definition can not be edited.
+In this panel, new node information such as node name, version, etc. can be specified. There is **Include All SUBFLOW** check box on this panel.  If this is checked, SUBFLOWs used within the exported SUBFLOW are recursively included in exported SUBFLOW Module.  If **Editable** check box on this panel is checked, SUBFLOW template can be edited on imported Node-RED editor.  Otherwise, the imported SUBLOW template is sealed. Thus, its internal flow or UI definition can not be edited.
```

Comment:
I think *most* of this extra information should be on a new tab in the subflow *edit* dialog that is saved as properties of the subflow regardless of the Export work flow.

If I am the author of a subflow that I want to distribute, I will have a local Node-RED project where I develop and test the subflow. I then export it and share with others.
If I then need to make an update to the subflow (such as a bug fix), I do not want to have to re-enter all of these extra properties when I re-export. That could easily lead to a mistake being made. I should be able to reopen my Node-RED project, make the change to the subflow and re-export.

Reply:
Moved properties edit entries to SUBFLOW Template properties edit tab.